assets/src/scripts/selectors/timeSeriesSelector.js

```diff
@@ -39,37 +39,40 @@ export const getCurrentParmCd = createSelector(
-};
 /*
- * @param {String} tsKey - current, compare, or median
- * @param {String} or null period - date range of interest specified as an ISO-8601 duration. If null P7D is assumed
- * @param {String} or null parmCD - Only need to specify if period is something other than P7D or null
- * @return {Boolean} - True if the time series with key, period, and parmCd has already been requested
- *
- */
-export const hasTimeSeries = memoize((tsKey, period, parmCd) => {
-    const periodToUse = period ? period : 'P7D';
-    const requestKey = tsRequestKey(tsKey, periodToUse, parmCd);
-    return state => state.series && state.series.requests && state.series.requests[requestKey] ? true : false;
-});
-
-/*
- * @param {String} tsKey - current, compare, or median
- * @param {String} or null - period to use, otherwise the current date range is used.
- * @return {String} or null - Return the the request key for the request object for the tsKey and period and currently
+ * @param {String} - time series key: current, compre or median
+ * @param {String} or null period = date range of interest as an ISO-8601 duration. If null the currentDateRange is used
+ * @param {String} or null parmCD - if null the parmCd of the current variable is used.
+ * @return {String} or null - Return the the request key for the request object
  * selected variable.
  */
-export const getCurrentVariableTimeSeriesRequestKey = memoize((tsKey, period )=> createSelector(
-    getCurrentParmCd,
+export const getTsRequestKey = memoize((tsKey, period, parmCd) => createSelector(
     getCurrentDateRange,
-    (parmCd, currentPeriod) => {
-        if (parmCd && (period || currentPeriod)) {
-            const periodToUse = period ? period : currentPeriod;
-            return tsRequestKey(tsKey, periodToUse, parmCd);
-        } else {
-            return null;
+    getCurrentParmCd,
+    (dateRange, currentParmCd) => {
+        let result = `${tsKey}`;
+        if (tsKey !== 'median') {
+            const periodToUse = period ? period : dateRange;
+            result += `:${periodToUse}`;
+            if (periodToUse !== 'P7D') {
+                const parmCdToUse = parmCd ? parmCd : currentParmCd;
+                result += `:${parmCdToUse}`;
+            }
         }
+        return result;
     })
 );
+/*
+ * @param {String} tsKey - current, compare, or median
+ * @param {String} or null period - date range of interest specified as an ISO-8601 duration. If null, P7D is assumed
+ * @param {String} or null parmCD - Only need to specify if period is something other than P7D or null
+ * @return {Boolean} - True if the time series with key, period, and parmCd has already been requested
+ *
+ */
+export const hasTimeSeries = memoize((tsKey, period, parmCd) => createSelector(
+    getTsRequestKey(tsKey, period, parmCd),
+    state => state.series,
+    (tsRequestKey, series) => {
+        return series && series.requests && series.requests[tsRequestKey] ? true : false;
```

Comment:
Here you could omit the `? true : false` part

Reply:
True, but then the result will be "truthy" or "falsy" not true or false. I may be being to strict about this but the comment says its returning a boolean so I'm returning a Boolean

Reply:
Ah, okay. Maybe `Boolean(...)`?

Reply:
👍